crates/forge_domain/src/orch.rs

```diff
@@ -515,21 +515,18 @@ impl<A: Services> Orchestrator<A> {
             if empty_tool_calls {
                 // No tool calls present, which doesn't mean task is complete so reprompt the
                 // agent to ensure the task complete.
-                let content = self
-                    .services
-                    .template_service()
-                    .render("{{> partial-tool-required.hbs}}", &())?;
+                let content = self.services.template_service().render(
+                    "{{> partial-tool-required.hbs}}",
+                    &serde_json::json!({
+                        "tool_supported": agent.tool_supported.unwrap_or_default()
```

Comment:
Add a new method in `ProviderService::model(&self, model: &ModelId) -> Result<Option<Model>>;`.
If the request fails with propagate the error, if there is a None, we use `agent.tool_supported`setting